kloppy/domain/services/transformers/dataset.py

```diff
@@ -196,57 +197,63 @@ def transform_frame(self, frame: Frame) -> Frame:
-            frame_id=frame.frame_id,
-            ball_owning_team=frame.ball_owning_team,
-            ball_state=frame.ball_state,
-            period=frame.period,
-            # changes
-            ball_coordinates=self.__change_point_coordinate_system(
-                frame.ball_coordinates
-            ),
-            ball_speed=frame.ball_speed,
-            players_data={
-                key: PlayerData(
-                    coordinates=self.__change_point_coordinate_system(
-                        player_data.coordinates
-                    ),
-                    distance=player_data.distance,
-                    speed=player_data.speed,
-                    other_data=player_data.other_data,
-                )
-                for key, player_data in frame.players_data.items()
-            },
-            other_data=frame.other_data,
+        frame = create_frame(
+            **dict(
+                # doesn't change
+                timestamp=frame.timestamp,
+                frame_id=frame.frame_id,
+                ball_owning_team=frame.ball_owning_team,
+                ball_state=frame.ball_state,
+                period=frame.period,
+                # changes
+                ball_coordinates=self.__change_point_coordinate_system(
+                    frame.ball_coordinates
+                ),
+                ball_speed=frame.ball_speed,
+                players_data={
+                    key: PlayerData(
+                        coordinates=self.__change_point_coordinate_system(
+                            player_data.coordinates
+                        ),
+                        distance=player_data.distance,
+                        speed=player_data.speed,
+                        other_data=player_data.other_data,
+                    )
+                    for key, player_data in frame.players_data.items()
+                },
+                other_data=frame.other_data,
+            )
         )
+        return frame
 
     def __change_frame_dimensions(self, frame: Frame):
-        return Frame(
-            # doesn't change
-            timestamp=frame.timestamp,
-            frame_id=frame.frame_id,
-            ball_owning_team=frame.ball_owning_team,
-            ball_state=frame.ball_state,
-            period=frame.period,
-            # changes
-            ball_coordinates=self.change_point_dimensions(
-                frame.ball_coordinates
-            ),
-            players_data={
-                key: PlayerData(
-                    coordinates=self.change_point_dimensions(
-                        player_data.coordinates
-                    ),
-                    distance=player_data.distance,
-                    speed=player_data.speed,
-                    other_data=player_data.other_data,
-                )
-                for key, player_data in frame.players_data.items()
-            },
-            other_data=frame.other_data,
+        frame = create_frame(
+            **dict(
```

Comment:
The factory shouldn't be used here. The `statistics` needs to be passed here to `Frame`